### PR TITLE
Enable client snippet before the language server is ready

### PR DESCRIPTION
**Pull request by jdneo**
Since we have moved some snippets from the client side to server side. The moved snippets will not be available until the language server is ready to provide the completion items.

This PR is trying to resolve this problem. It will register a completion provider to provide those snippets which has been moved to the server side. Once the client received the notification that the language server is ready, the client side provider will be disposed.

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: src/snippetCompletionProvider.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+import { CompletionItemProvider, TextDocument, Position, CancellationToken, CompletionContext, CompletionItem, CompletionItemKind, SnippetString, MarkdownString } from "vscode";
+import * as fse from 'fs-extra';
+import * as path from 'path';
+
+export class SnippetCompletionProvider implements CompletionItemProvider {
+    public async provideCompletionItems(_document: TextDocument, _position: Position, _token: CancellationToken, _context: CompletionContext): Promise<CompletionItem[]> {
+        const snippetItems: CompletionItem[] = [];
+        const snippets: {} = await fse.readJSON(path.join(__dirname, '..', 'snippets', 'server.json'));
```
Comment (fbricon) [inline src/snippetCompletionProvider.ts]: can we avoid rereading the file every time completion is called?

**Comment by jdneo** [comment]
No need to be included into today's release. Since I believe this is not addressing something blocking. -> my personal thinking.

**Comment by jdneo** [comment]
Hmm, where can I find the copyright headers? I cannot find one in others ts files.

File: src/snippetCompletionProvider.ts (hunk 1 above)
Comment (jdneo) [reply src/snippetCompletionProvider.ts]: Sorry I should put it into the constructor

**Comment by fbricon** [comment]
Can you please fix the merge conflict?

**Comment by fbricon** [comment]
@jdneo can you please fix the conflict?

**Comment by jdneo** [comment]
Sorry I just notice the comment. Sure I'll resolve them ASAP.

**Comment by fbricon** [comment]
@jdneo sorry but another commit caused a new conflict

**Comment by jdneo** [comment]
No problem, it's updated now

**Pull request merged**